### ISSUE TITLE
Component: Comparison Chart

### ISSUE THREAD

**Issue by LevinMedia** [woocommerce/wc-admin#695]
**Blocked: ** Until the API can be segmented, all results are lumped together. When that happens, `ReportChart` [can be modified](https://github.com/woocommerce/wc-admin/blob/b945875fae27f44e7a49d9b12347ab007cf2ca4c/client/analytics/components/report-chart/index.js#L66-L76) to deliver the segments to the chart:

The comparison chart differs from a standard chart (IE, what you'll see in the revenue report) in that it allows a user to compare (surprise!) multiple series of data. it's detailed at: p6riRB-3u4-p2

**The core reports that will feature the comparison chart in addition to the standard chart include:**
Product / Product Detail Report
Category Report
Coupon Report
Tax Report

**Comparison chart in the context of the product detail report:**

<img width="1172" alt="screen shot 2018-10-25 at 7 07 58 pm" src="https://user-images.githubusercontent.com/4500952/47540302-5b60f480-d889-11e8-8f15-9ad4f0e1c142.png">

- Data points selected for comparison appear in the legend to the left of the chart
- A maximum of (X) data points can be plotted in the graph at any given time, however, an unlimited number of data points can be added to the comparison (The final number of data points to plot at any give time are still TBD, and depend on two factors: load performance, and visual design)
- As data points are added to the comparison, the legend will eventually overflow its container and become scrollable vertically. 
- To provide context, the number of data points being compared are displayed at the bottom of the legend.  

**A variation on the comparison chart: Filtering a report by top products, top coupons etc** 

A select menu is introduced to the chart header which allows a user to pick the number of data points to compare. Further detail at p6riRB-3jq-p2

<img width="711" alt="screen shot 2018-10-25 at 7 32 55 pm" src="https://user-images.githubusercontent.com/4500952/47541159-1e96fc80-d88d-11e8-91e9-51518a4d8852.png">

In all cases where a comparison chart is being used, the summary numbers and tabular data update to reflect only what's being compared.

**Comment by psealock** [comment]
Information on the progress of segmenting the API response in p7bje6-1bY-p2

**Comment by justinshreve** [comment]
> In all cases where a comparison chart is being used, the summary numbers and tabular data update to reflect only what's being compared.

I wanted to double check about this @LevinMedia.

So with all variations selected it would look like this:

<img width="972" alt="screen shot 2019-01-21 at 10 24 34 pm" src="https://user-images.githubusercontent.com/689165/51510717-34548900-1dcc-11e9-9556-d7f5d04dd481.png">

<img width="814" alt="screen shot 2019-01-21 at 10 27 15 pm" src="https://user-images.githubusercontent.com/689165/51510725-420a0e80-1dcc-11e9-8208-3edf186c2fdb.png">

After I unselected on of these in the chart, the following happens (notice that the summary numbers and table update):

<img width="819" alt="screen shot 2019-01-21 at 10 29 35 pm" src="https://user-images.githubusercontent.com/689165/51510747-5bab5600-1dcc-11e9-89a5-e023fc9912c9.png">

<img width="808" alt="screen shot 2019-01-21 at 10 28 36 pm" src="https://user-images.githubusercontent.com/689165/51510753-60700a00-1dcc-11e9-975d-14f3f677f2c9.png">

---

Is that correct? It seems weird to me that the unselected values would be removed from the table. Or would they remain in the table and just be unchecked (with the summary numbers in the footer and header reflecting only the selected values)?

**Comment by LevinMedia** [comment]
>Is that correct? It seems weird to me that the unselected values would be removed from the table.

Heya @justinshreve - your instinct is correct. In this case, If a series is unchecked in the chart, it should not be removed from the table. The summary numbers shouldn't change either., they should always calculate aggregate the totals for all of the values being compared.  

For example, someone may want to compare 50 coupons at once, or view a single product with 50 variations - in those cases, the summary numbers and tabular data will aggregate data for all of the  selected coupons or all the variations from the product. For performance reasons, we'll only be able to render a comparison of 8 to 10 series in the chart at one time (still TBD). The first sketch at the top of this issue illustrates that senario. A user could check and uncheck various series to compare how different values have performed over time. 

<img width="671" alt="screen shot 2019-01-21 at 7 46 00 pm" src="https://user-images.githubusercontent.com/4500952/51511383-ecc30280-1db5-11e9-9e0d-dbf5d406a67c.png">

**Comment by LevinMedia** [comment]
@justinshreve et al. wanted to point it out because I saw it in the first screenshots you posted: 

<img width="669" alt="screen shot 2019-01-21 at 8 01 40 pm" src="https://user-images.githubusercontent.com/4500952/51511772-6c050600-1db7-11e9-8a73-8e9bd24af647.png">


By default when viewing a single product, or after a user has made a comparison, the rows in the tabular data should no longer be checked. In the example I gave above, if a someone wanted to hone in on a more granular comparison of only a couple coupons or variations, they could check off a couple rows in the table, and compare again.
